thecourseforum/theCourseForum2
# Coverage precommit

## Status
- Done

## GitHub Issues addressed (`#IssueNumber`)
- This PR closes #361

## What I did
- Added `coverage` to `precommit` so that developers can check Django code coverage locally

## Screenshots
- Before
![image](https://user-images.githubusercontent.com/7676354/107044064-d10ad900-6791-11eb-93f1-28fa3ce90796.png)
- After
![image](https://user-images.githubusercontent.com/7676354/107043765-71143280-6791-11eb-9fe1-808f6149541c.png)

## Tests
- Run `docker-compose up` and check, in the docker log, if `coverage` is installed.
```
tcf_django | Collecting coverage
tcf_django |   Downloading coverage-5.4-cp38-cp38-manylinux2010_x86_64.whl (245 kB)- 
```
- Open another terminal--a Unix shell--and run `./precommit` and verify that there's a coverage report at the end.

## Questions/Discussions/Notes
- None

## Merging the PR
- Who should merge this PR?
  - Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
